src/tiled/scriptfileformatwrappers.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * scriptfileformatwrappers.h
+ * Copyright 2019, Thorbjørn Lindeijer <bjorn@lindeijer.nl>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "scriptfileformatwrappers.h"
+
+#include "editablemap.h"
+#include "editabletileset.h"
+#include "mapformat.h"
+#include "scriptedfileformat.h"
+#include "scriptmanager.h"
+#include "tilesetformat.h"
+
+#include <QCoreApplication>
+#include <QQmlEngine>
+
+namespace Tiled {
+
+ScriptTilesetFormatWrapper::ScriptTilesetFormatWrapper(TilesetFormat* format, QObject *parent)
+    : QObject(parent)
+    , mFormat(format)
+{}
+
+QString ScriptTilesetFormatWrapper::name() const
+{
+    if (auto scriptedFormat = qobject_cast<ScriptedTilesetFormat*>(mFormat))
+        return scriptedFormat->name();
+    else
+        return mFormat->shortName();
+}
+
+QString ScriptTilesetFormatWrapper::extension() const
+{
+    if (auto scriptedFormat = qobject_cast<ScriptedTilesetFormat*>(mFormat))
+        return scriptedFormat->extension();
+    else
+        return QString();
+}
+
+EditableTileset *ScriptTilesetFormatWrapper::read(QString &filename)
+{
+    auto tileset = mFormat->read(filename);
+    auto editable = new EditableTileset(tileset.data());
```

Comment:
`tileset` may be a null shared pointer, and the possible error should probably be thrown.